gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/DefaultKeystoreService.java

```diff
@@ -513,12 +513,14 @@ private synchronized boolean isKeyStoreAvailable(final Path keyStoreFilePath, St
   // Package private for unit test access
   // We need this to be synchronized to prevent multiple threads from using at once
   synchronized KeyStore createKeyStore(Path keystoreFilePath, String keystoreType, char[] password) throws KeystoreServiceException {
-    if (Files.notExists(keystoreFilePath)) {
-      // Ensure the parent directory exists...
-      try {
+    // Ensure the parent directory exists...
+    // This is symlink safe.
+    Path parentPath = keystoreFilePath.getParent();
```

Comment:
@jameschen1519 - it is possible for this method to return null, I believe.
I am not sure what the implications on the next line would be in that case,
I think that you should extend the existing test cases in DefaultKeystoreServiceTest to provide proper assurances here.

Reply:
Thanks for the catch. I think it would be sufficient to just add a null check to the if statement, since we want the call to getParent to be successful when the keystore directory happens to be the root directory.

I'll try adding in a test case to check if this fix works if the last directory is a symlink.

Reply:
Test has been added.